elastic/kibana
# [Time to Visualize] Fix Unlink Action via Rollback of ReplacePanel

## Summary

Fixes https://github.com/elastic/kibana/issues/83864 via a partial rollback of https://github.com/elastic/kibana/pull/74253.

The new version of the Replace Panel method introduced in that PR _always_ retains Embeddable Ids through the replacement process. This avoids some flickering, which was especially prevalent in the clone function. Unfortunately, with Actions like `Unlink From Library` and `Add to Library` which add, remove, or update savedObjectIds, retaining the same `EmbeddableId` prevents the Embeddable from fully re-initializing with the new SavedObjectId.

In order to fix this, we can either make all `by value / by reference` embeddables respond properly to changes in their savedObjectId - POC PR for Lens https://github.com/elastic/kibana/pull/83874 -

OR we can force a total re-initialization for Embeddables in these special cases where SavedObjectIds change. The latter is what this PR does.

#### How to Test
1. Add `dashboard.allowByValueEmbeddables: true` to the bottom of your `config/kibana.dev.yml`
2. Load a dashboard, and add a Lens panel to it from the Visualize Library
3. Use the 'unlink from library' action to unlink the panel from the library
4. Edit the newly unlinked panel. You should _not_ see a savedObjectId in the Url bar, and the address should contain edit_by_value.

This same procedure should work correctly with Visualize and Maps as well.


### Checklist

Delete any items that are not applicable to this PR.

- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or 

added to match the most common scenarios
### For maintainers
